data_diff/config.py

```diff
@@ -0,0 +1,73 @@
+from typing import Any, Dict
+import toml
+
+
+class ConfigParseError(Exception):
+    pass
+
+
+def is_uri(s: str) -> bool:
+    return "://" in s
+
+
+def _apply_config(config: Dict[str, Any], run_name: str, kw: Dict[str, Any]):
+    # Load config
+    databases = config.pop("database", {})
+    runs = config.pop("run", {})
+    if config:
+        raise ConfigParseError(f"Unknown option(s): {config}")
+
+    # Init run_args
+    run_args = runs.get("default") or {}
+    if run_name:
+        if run_name not in runs:
+            raise ConfigParseError(f"Cannot find run '{run_name}' in configuration.")
+        run_args.update(runs[run_name])
+    else:
+        run_name = "default"
+
+    # Process databases + tables
+    for index in "12":
+        args = run_args.pop(index, {})
+        for attr in ("database", "table"):
+            if attr not in args:
+                raise ConfigParseError(f"Running 'run.{run_name}': Connection #{index} in missing attribute '{attr}'.")
+
+        database = args.pop("database")
+        table = args.pop("table")
+        threads = args.pop("threads", None)
+        if args:
+            raise ConfigParseError(f"Unexpected attributes for connection #{index}: {args}")
+
+        if not is_uri(database):
+            if database not in databases:
+                raise ConfigParseError(
+                    f"Database '{database}' not found in list of databases. Available: {list(databases)}."
+                )
+            database = dict(databases[database])
+            assert isinstance(database, dict)
+            if "driver" not in database:
+                raise ConfigParseError(f"Database '{database}' did not specify a driver.")
+
+        run_args[f"database{index}"] = database
+        run_args[f"table{index}"] = table
+        if threads is not None:
+            run_args[f"threads{index}"] = int(threads)
+
+    # Update keywords
+    new_kw = dict(kw)  # Set defaults
+    new_kw.update(run_args)  # Apply config
+    new_kw.update({k: v for k, v in kw.items() if v})  # Apply non-empty defaults
```

Comment:
I am not sure how this `if v` works here: If the new value of `verbose` is False (e.g., due to a hypothetical `--quiet` flag), will it override the TOML setting or be ignored? The same for values that are int 0, if any. Would it make more sense only to exclude `if v is not None` — i.e. the special marker of emptiness?

Reply:
The default for `click` is False, not None. I don't control the default values, I just define it as `is_flag`.

Yes, for a `--quiet` flag I'll have to change the way it works. Either get rid of `click`, or more likely, define the parameters in our own format, and translate that for click.

Reply:
I see. It makes sense now: so, the default False should not override the config value. Thanks for clarifying.